### DISCUSSION TITLE
Add memory profiling for catalog

### DISCUSSION THREAD

**Discussion by noklam** [kedro-org/kedro#4493] [Idea]
## Context 

In the current setup, we use the same pipeline and catalog configuration and we try to test memory usage by running nodes with heavy compute load that should not grow the memory. So it feels like we might add other scenarios that grow it and we're able to compare (memory grows as expected) runs on them.

_Originally posted by @ElenaKhaustova in https://github.com/kedro-org/kedro/pull/4210#discussion_r1819227669_

## Description
Add memory profiling for catalogs.

**Comment by astrojuanlu** [comment]
> Add memory profiling for catalogs.

What's the ask here? Can't users use memory-profiler or https://github.com/bloomberg/memray/ with Kedro at the moment?
